### PR TITLE
Allow markdown for content pages

### PR DESCRIPTION
**Pull request by barbarah**
This is the technical part of this ticket colonial-heritage/sprints#90 

I decided to use Markdown for these pages because the content is long and includes markup. Before I translate the word document to Markdown, I would like to know if this is the correct way to go.

I followed this guide: https://beta.nextjs.org/docs/guides/mdx

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +24,9 @@
     "@hapi/hoek": "11.0.2",
     "@headlessui/react": "1.7.9",
     "@heroicons/react": "2.0.16",
+    "@next/mdx": "^13.2.4",
     "@tailwindcss/forms": "0.5.3",
+    "@types/mdx": "^2.0.3",
```
Comment (sdevalk) [inline package.json:29]: Should this be a part of `devDependencies`?

File: package.json [hunk 2]
```diff
@@ -24,7 +24,9 @@
     "@hapi/hoek": "11.0.2",
     "@headlessui/react": "1.7.9",
     "@heroicons/react": "2.0.16",
+    "@next/mdx": "^13.2.4",
```
Comment (sdevalk) [inline package.json:27]: For consistency: remove the `^`?

File: src/app/[locale]/about/page.tsx [hunk 3]
```diff
@@ -1,12 +1,8 @@
-import {useTranslations} from 'next-intl';
-import {PageHeader, PageTitle} from '@/components/page';
+import {useLocale} from 'next-intl';
 
-export default function About() {
-  const t = useTranslations('About');
+export default async function About() {
+  const locale = useLocale();
+  const AboutPage = (await import(`@/messages/${locale}/about.mdx`)).default;
```
Comment (sdevalk) [inline src/app/[locale]/about/page.tsx:5]: Curious: Is there a way to not have to use the `.default`?

**Review by sdevalk — COMMENTED** [review]
Great idea to use Markdown - I like it! 👍🏻

File: src/app/[locale]/about/page.tsx (hunk 3 above)
Comment (barbarah) [reply src/app/[locale]/about/page.tsx:5]: I don't think so. But I have moved this logic to one LocalizedMarkdown component, so it is simpler to use.

File: src/components/localized-markdown.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    notFound();
+  }
+}
+
+// TypeScript doesn't understand async components yet.
+// So this is a temporary workaround.
+// More info:
+//  - Next.js issue: https://github.com/vercel/next.js/issues/42292
+//  - Typescript pull request: https://github.com/microsoft/TypeScript/pull/51328
+export default LocalizedMarkdown as unknown as ({name}: Props) => JSX.Element;
```
Comment (sdevalk) [inline src/components/localized-markdown.tsx:23]: Curious: do we need to be explicit here (`({name}: Props)`)? Or can we just refer to the type: `(props: Props)`?

**Review by sdevalk — COMMENTED** [review]
Nice! Just one question.

**Review by sdevalk — APPROVED** [review]
Cool!

**Pull request merged**